cgns: Add -fPIC to Fortan/C compilation flags

payerle opened the pull request:
This should fix #39699

Adds -fPIC to C/Fortran compilation flags

payerle committed:
cgns: Add -fPIC to Fortan/C compilation flags

This should fix #39699

aumuell commented on var/spack/repos/builtin/packages/cgns/package.py:
```diff
@@ -83,6 +83,8 @@ def cmake_args(self):
                 self.define_from_variant("CGNS_ENABLE_BASE_SCOPE", "base_scope"),
                 self.define_from_variant("CGNS_ENABLE_LEGACY", "legacy"),
                 self.define_from_variant("CGNS_ENABLE_MEM_DEBUG", "mem_debug"),
+                "-DCMAKE_Fortran_FLAGS=-fPIC",
```
I don't know about Fortran, but wouldn't it be more portable to let CMake add the appropriate flags:
  `self.define_from_variant("CMAKE_POSITION_INDEPENDENT_CODE", "pic"),`
And perhaps following a variant?
  `variant("pic", default=False, description="Produce position-independent code")`

payerle replied:
Thank you for your suggestions.  I have updated the patch to incorporate them, with the addition of a conflict to require +pic when +fortran (as +fortran ~pic will trigger #39699

payerle committed:
cgns: Add -fPIC compilation flags when +fortran

Incorporating suggestions of @aumuell
with addition of making +pic required when +fortran

tldahlgren reviewed (approved):
LGTM
